File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:


        def operator(op):
            b = st.pop()
            a = st.pop() 

            if op=="+":
                st.append(a+b)
            elif op=="-":
                st.append(a-b)
            elif op=="*":
                st.append(a*b)
            else: 
                st.append(int(a/b)) 


        st = [] 
        for t in tokens: 
            if t.isdigit() or len(t)>=2:
                st.append(int(t)) 
            else: 
                operator(t) 

        return st[-1]

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        
        stack<long long> st ; 
        for(auto& t:tokens){
            if( t=="+" || t=="-" || t== "*" || t =="/"){
                long long a = st.top() ; st.pop() ; 
                long long  b = st.top() ; st.pop() ; 
                if(t=="+") st.push(b+a) ;
                if(t=="-") st.push(b-a) ; 
                if(t=="*") st.push(b*a) ; 
                if(t=="/") st.push(b/a) ;
            }else {
                st.push(stol(t));
            }
        }
        return st.top() ; 


    }
};